### PR TITLE
Manage users

### PR DESCRIPTION
**Pull request by evgeny-sav**

### CODE REVIEW

File: notes_app/client/notes.js [hunk 1]
```diff
@@ -54,6 +57,17 @@
   angular.module('notes').run(function($rootScope, $state, userService) {
     $rootScope.$on('$stateChangeStart',
       function(event, toState, toParams, fromState, fromParams, options) {
+          (function() {
+            var promise = new Promise(function(resolve, reject){
```
Comment (dglebovskiy) [inline notes_app/client/notes.js]: Aren't we supposed to be using Angular's implementation of promises ($q)?

Comment (evgeny-sav) [reply notes_app/client/notes.js]: I have just done it. You will see it in next commit :)

File: notes_app/client/app/users/users_list.component.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+         ctrl.rowsToDelete = [];
+         ctrl.gridOptions = {
+            data: [],
+            paginationPageSize: 5,
+            paginationPageSizes: [5,10,25],
+            enableRowHeaderSelection: true,
+            multiSelect: true,
+            enableSelectAll: false,
+            columnDefs: [
+            {name: 'id', enableCellEdit: false, width: '5%' },
```
Comment (ghost) [inline notes_app/client/app/users/users_list.component.js]: @dglebovskiy @evgeny-sav do you use some linters and code style checkers guys?

File: notes_app/client/app/users/user_service.js [hunk 3]
```diff
@@ -15,6 +15,24 @@
           return [];
         });
       },
+      'update': function(id, data) {
+        return $http.put(api_path + id + '/', data).then(function(response){
+            return;
+        }).catch(function(error){
+            return $q.reject({
+                'error': error
```
Comment (ghost) [inline notes_app/client/app/users/user_service.js]: what is the point of new object here?

File: notes_app/client/app/users/user_service.js [hunk 4]
```diff
@@ -15,6 +15,24 @@
           return [];
         });
       },
+      'update': function(id, data) {
+        return $http.put(api_path + id + '/', data).then(function(response){
+            return;
```
Comment (ghost) [inline notes_app/client/app/users/user_service.js]: here is 4 spaces indentation, it's not solid with the rest code

File: notes_app/client/app/users/users_list.component.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+            {name: 'first_name', enableCellEdit: true},
+            {name: 'last_name', enableCellEdit: true},
+            {name: 'username', enableCellEdit: true},
+            {name: 'email', field: 'email', enableCellEdit: true},
+            {name: 'is_superuser', enableCellEdit: false, width: '10%' },
+            ],
+         };
+
+
+         ctrl.saveRow = function( rowEntity ) {
```
Comment (ghost) [inline notes_app/client/app/users/users_list.component.js]: is it OK by code style?

File: notes_app/client/notes.js [hunk 6]
```diff
@@ -45,9 +58,20 @@
...
+  angular.module('notes').run(function($rootScope, $state, userService, $q) {
     $rootScope.$on('$stateChangeStart',
       function(event, toState, toParams, fromState, fromParams, options) {
+          (function() {
+            var promise = $q(function(resolve, reject){
+                resolve(userService.user.data);
+            });
+            return promise;
+          })().then(function(res){
+            if (toState.url === 'users' && !res.is_superuser ) {
```
Comment (ghost) [inline notes_app/client/notes.js]: so if you need other superusers-only states later you'll add more `if`s? Doesn't look handy to me. Check uiRouter docs for `onEnter` state property

File: notes_app/client/templates/user.tpl.html [hunk 7]
```diff
@@ -0,0 +1,2 @@
+
```
Comment (ghost) [inline notes_app/client/templates/user.tpl.html]: just empty file? what is the point to have it?

File: notes_app/client/templates/users_list.tpl.html [hunk 8]
```diff
@@ -0,0 +1,25 @@
+<div>
+    <style>
+        .ui-grid-header-cell {
+            height: 30px;
+        }
+    </style>
+    <div
+            ui-grid="$ctrl.gridOptions"
+
```
Comment (ghost) [inline notes_app/client/templates/users_list.tpl.html]: tooooooooooooooo maaaaaaaaaaaaaaaanyyyyyyyyyyyyy empty lines, it's harder to read. And it's a very big indentation - 4 spaces - in html.

**Review by ghost — APPROVED** [review]

**Pull request merged**